tests/src/testassembler.cpp

```diff
@@ -84,30 +85,19 @@ auto SimpleAssemblersTest(const PreBasis& preBasis) {
     d.setRandom();
     double load = 0.0;
 
-    Ikarus::FERequirements req = Ikarus::FERequirements()
-                                     .insertGlobalSolution(Ikarus::FESolutions::displacement, d)
-                                     .insertParameter(Ikarus::FEParameter::loadfactor, load)
-                                     .addAffordance(Ikarus::AffordanceCollections::elastoStatics);
+    auto req = typename FEType::Requirement(d, load);
 
-    t.check(req.hasAffordance(Ikarus::MatrixAffordances::stiffness));
-    t.check(not req.hasAffordance(Ikarus::MatrixAffordances::mass));
-    t.check(req.hasAffordance(Ikarus::ScalarAffordances::mechanicalPotentialEnergy));
-    t.check(req.hasAffordance(Ikarus::VectorAffordances::forces));
-
-    Ikarus::FERequirements req2 = Ikarus::FERequirements().addAffordance(Ikarus::mass);
-    t.check(req2.hasAffordance(Ikarus::MatrixAffordances::mass));
-
-    auto& KRawDense = denseFlatAssembler.getRawMatrix(req);
-    auto& KRaw      = sparseFlatAssembler.getRawMatrix(req);
-    auto& RRawDense = denseFlatAssembler.getRawVector(req);
-    auto& RRaw      = sparseFlatAssembler.getRawVector(req);
+    auto& KRawDense = denseFlatAssembler.matrix(req, MatrixAffordance::stiffness, EnforcingDBCOption::Raw);
+    auto& KRaw      = sparseFlatAssembler.matrix(req, MatrixAffordance::stiffness, EnforcingDBCOption::Raw);
+    auto& RRawDense = denseFlatAssembler.vector(req, VectorAffordance::forces, EnforcingDBCOption::Raw);
+    auto& RRaw      = sparseFlatAssembler.vector(req, VectorAffordance::forces, EnforcingDBCOption::Raw);
     checkAssembledQuantities(t, KRaw, KRawDense, totalDOFs);
     checkAssembledQuantities(t, RRaw, RRawDense, totalDOFs);
 
-    auto& KDense = denseFlatAssembler.getMatrix(req);
-    auto& K      = sparseFlatAssembler.getMatrix(req);
-    auto& RDense = denseFlatAssembler.getVector(req);
-    auto& R      = sparseFlatAssembler.getVector(req);
+    auto& KDense = denseFlatAssembler.matrix(req, MatrixAffordance::stiffness, EnforcingDBCOption::Full);
+    auto& K      = sparseFlatAssembler.matrix(req, MatrixAffordance::stiffness, EnforcingDBCOption::Full);
+    auto& RDense = denseFlatAssembler.vector(req, VectorAffordance::forces, EnforcingDBCOption::Full);
+    auto& R      = sparseFlatAssembler.vector(req, VectorAffordance::forces, EnforcingDBCOption::Full);
```

Comment:
may be we can also have here `vectorAffordance(MatrixAffordance::stiffness)` instead of `VectorAffordance::forces` to check if this works correctly? I am not sure if I overlooked and it is already tested somewhere else

Reply:
It is needed in autodiff. Thats was my main motivation to add it